elastic/kibana
# [Infrastructure UI] Simplify unified search usage

## Summary

closes [152574](https://github.com/elastic/kibana/issues/152574)
fixes [144970](https://github.com/elastic/kibana/issues/144970) (partially)
fixes [152546](https://github.com/elastic/kibana/issues/152546)

This PR simplifies how unified search is used in the Hosts View,  by using the SearchBar default behaviors and subscribing to its services observables instead of using its event handlers. 

Observables are not the most intuitive approach, but the `filterManager` service observable was already being subscribed to cover the integration with Lens. This PR standardizes it to only use observables

By doing that, some known problems as duplicate requests when clearing, saving, and loading filters were also solved. I couldn't really split up the solution to address each individual ticket.


### How to test
- Navigate to Infrastructure > Hosts
  - Set a filter, query, and date range and confirm if the data on the page reflects the search criteria
  - Verify if filtering through Lens charts works
  - Save a query (shouldn't trigger any request)
  - Load a saved query  (shouldn't make duplicate request)
  - Clear all search criteria (shouldn't make duplicate requests)
  - Copy a URL with filters and open it in another tab

### Note to reviewer
There is still one strange scenario in which clearing all may or may not make duplicate requests: 
- Select a filter in the filter control
- Add a filter in the Unified Search bar
- Save this filter
- Clear all
